src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/index.jsx

```diff
@@ -301,15 +301,19 @@ class GlobalSelectionHeader extends React.Component {
   };
 
   getProjects = () => {
+    const {projects} = this.props;
     const {isSuperuser} = ConfigStore.get('user');
-    const {projects, organization} = this.props;
-    const unfilteredProjects = projects || organization.projects;
+
+    const [memberProjects, nonMemberProjects] = partition(
+      projects,
+      project => project.isMember
+    );
```

Comment:
Do we need to run the partition for all users? For everyone other than superusers `nonMemberProjects` will be empty. Could this be moved inside the `if (isSuperuser)` block to avoid CPU burn for other users?

Reply:
We were already filtering on the same function on line 312, so I don't think it really causes more CPU burn than before. Since we'll adding org admins and other potential non superuser users to this list soon I think it'd be good to leave it here.

Reply:
If non-superusers start being able to see projects they aren't a member of we should consider re-working where the global header gets its data from, and using a typeahead based solution. Having a picker with 100's of items isn't going to perform well.

Reply:
What do you mean by typeahead solution? You mean if we remove all projects from org details endpoint we should query on each keystroke? The dropdown is virtualized already and we do have a text filter so I think the UI performance won't be affected too much currently even if this list is long

Reply:
> You mean if we remove all projects from org details endpoint we should query on each keystroke? 

My thinking was around the weight of the API response. We've done what we can to make the rendering lightweight, but fetching the sometimes very long list of projects inside of org details is problematic. Having it as a separate request might be enough, but all of that is a discussion for later.